src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -24,29 +22,123 @@ class ThemePatch : ResourcePatch {
         val darkThemeBackgroundColor = darkThemeBackgroundColor!!
         val lightThemeBackgroundColor = lightThemeBackgroundColor!!
 
-        context.xmlEditor["res/values/colors.xml"].use { editor ->
-            val resourcesNode = editor.file.getElementsByTagName("resources").item(0) as Element
+        val valuesPaths: List<String> = listOf(
+            "values", // Android 11 (and below)
+            "values-v31" // Android 12 (and above)
+        )
+
+        // edit the resource color file to change the theme color
+        val colorsPath = "res/$valuesPaths[0]/colors.xml"
+
+        context.xmlEditor[colorsPath].use { editor ->
+            val childNodes = (editor.file.getElementsByTagName("resources").item(0) as Element).childNodes
 
-            for (i in 0 until resourcesNode.childNodes.length) {
-                val node = resourcesNode.childNodes.item(i) as? Element ?: continue
+            for (i in 0 until childNodes.length) {
+                val node = childNodes.item(i) as? Element ?: continue
 
                 node.textContent = when (node.getAttribute("name")) {
                     "yt_black0", "yt_black1", "yt_black1_opacity95", "yt_black1_opacity98", "yt_black2", "yt_black3",
                     "yt_black4", "yt_status_bar_background_dark", "material_grey_850" -> darkThemeBackgroundColor
 
-                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4",
-                     -> lightThemeBackgroundColor
+                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4"
+                    -> lightThemeBackgroundColor
 
                     else -> continue
                 }
             }
         }
 
-        // copies the resource file to change the splash screen color
-        context.copyResources("theme",
-            ResourceUtils.ResourceGroup("values-night-v31", "styles.xml")
+        // edit the resource files to change the splash screen color
+        val attrsPath = "res/$valuesPaths[0]/attrs.xml"
+        val drawablePaths: List<String> = listOf(
+            "drawable",
+            "drawable-anydpi",
+            "drawable-hdpi",
+            "drawable-ldrtl",
+            "drawable-ldrtl-night",
+            "drawable-ldrtl-xxhdpi",
+            "drawable-mdpi",
+            "drawable-night",
+            "drawable-nodpi",
+            "drawable-sw600dp",
+            "drawable-sw600dp-xhdpi",
+            "drawable-sw720dp-xhdpi",
+            "drawable-watch",
+            "drawable-xhdpi",
+            "drawable-xxhdpi",
+            "drawable-xxxhdpi"
         )
```

Comment:
Is `drawable-watch` even responsible for the splash screen? The same goes for the other resource directories. Are they all necessary?

Reply:
As I said on a previous request: I don't know. I simply added all the folder names you posted to me with a screenshot. On the apk that I managed to find only ```drawable``` and ```drawable-sw600``` contain the ```quantum_launchscreen``` file

Reply:
It is unclear if all of them are required. It depends on what the purpose of these separate resource directories is. The folders I screenshotted and their purpose should've separately been investigated to confirm whether they are required to be considered in this patch. For that, you will need an APK which is "anydpi".

Reply:
> It is unclear if all of them are required. It depends on what the purpose of these separate resource directories is. The folders I screenshotted and their purpose should've separately been investigated to confirm whether they are required to be considered in this patch. For that, you will need an APK which is "anydpi".

Wouldn't it be better to add these later, based on user feedback?

Reply:
I'll leave this request open until you decide how to proceed.